app/drivers/google/driver.go

```diff
@@ -394,21 +459,53 @@ func (p *config) DestroyInstanceAndStorage(ctx context.Context, instances []*typ
 			}
 		}
 
-		requestID := uuid.New().String()
-
-		_, err = p.deleteInstance(ctx, p.projectID, zone, instance.ID, requestID)
-		if err != nil {
+		instanceDeleteOperation, deleteInstanceErr := p.deleteInstance(ctx, p.projectID, zone, instance.ID, uuid.New().String())
+		if deleteInstanceErr != nil {
 			// https://github.com/googleapis/google-api-go-client/blob/master/googleapi/googleapi.go#L135
-			if gerr, ok := err.(*googleapi.Error); ok &&
+			if gerr, ok := deleteInstanceErr.(*googleapi.Error); ok &&
 				gerr.Code == http.StatusNotFound {
-				logr.WithError(err).Errorln("google: VM not found")
+				logr.WithError(deleteInstanceErr).Errorln("google: VM not found")
 			} else {
-				logr.WithError(err).Errorln("google: failed to delete the VM")
+				logr.WithError(deleteInstanceErr).Errorln("google: failed to delete the VM")
 			}
 		}
+		err = deleteInstanceErr
 		logr.Info("google: sent delete instance request")
+
+		if storageCleanupType != nil && *storageCleanupType == storage.Delete && instance.StorageIdentifier != "" {
+			logr.Info("google: waiting for instance deletion")
+			err = p.waitZoneOperation(ctx, instanceDeleteOperation.Name, zone)
+			if err != nil {
+				logr.WithError(err).Errorln("google: could not delete instance. skipping disk deletion")
+				return err
+			}
+			logr.Info("google: deleting persistent disk")
+			storageIdentifiers := strings.Split(instance.StorageIdentifier, ",")
+			for _, storageIdentifier := range storageIdentifiers {
+				diskDeleteOperation, diskDeletionErr := p.deletePersistentDisk(
+					ctx,
+					p.projectID,
+					zone,
+					storageIdentifier,
+					uuid.New().String(),
+				)
+				if diskDeletionErr != nil {
+					var googleErr *googleapi.Error
+					if errors.As(diskDeletionErr, &googleErr) &&
+						googleErr.Code == http.StatusNotFound {
+						logr.WithError(diskDeletionErr).
+							Errorln("google: persistent disk %s not found", storageIdentifier)
+					}
+				}
+				err = p.waitZoneOperation(ctx, diskDeleteOperation.Name, zone)
```

Comment:
Can diskDeleteOperation be nil if diskDeletionErr != nil ?

Reply:
It should not be. We are doing similar operations already for instance creation and deletion. Only in case of createPersistentDiskIfNotExists we are doing custom handling were operation can be null